Maybe.Test/MaybeTests.cs

```diff
@@ -328,5 +329,45 @@ public void OrMaybe_Func_MaybeNullArgument_ShouldThrow()
             subject.Should().ThrowExactly<ArgumentNullException>();
         }
 
+        [Fact]
+        public void ImplicitConversion_FromNullObject_ShouldReturnMaybeWithoutValue()
+        {
+            Maybe<object> subject = null;
+
+            subject.Should().Be(Maybe<object>.Nothing);
+        }
+
+        [Theory]
+        [MemberData(nameof(NonNullData))]
+        public void ImplicitConversion_FromNonNullObject_ShouldReturnMaybeWithValue<T>(T value)
+        {
+            Maybe<T> subject = value;
+
+            subject.Value.Should().Be(value);
+        }
+
+        public static TheoryData<object> NonNullData()
+        {
+            return new TheoryData<object>()
+            {
+                '1',
+                1,
```

Comment:
Let's add a test with a `0` so we can ensure that `if (value == null)` line doesn't get changed to a `if (value == default)`.